crates/services/p2p/src/codecs/postcard.rs

```diff
@@ -85,28 +82,36 @@ impl RequestResponseCodec for PostcardCodec {
 
     async fn read_request<T>(
         &mut self,
-        _protocol: &Self::Protocol,
+        _: &Self::Protocol,
         socket: &mut T,
     ) -> io::Result<Self::Request>
     where
         T: AsyncRead + Unpin + Send,
     {
-        let encoded_data = read_length_prefixed(socket, MAX_REQUEST_SIZE).await?;
+        let mut response = Vec::new();
+        socket
+            .take(self.max_response_size as u64)
```

Comment:
Is there a way to improve the error message reporting if `max_response_size` is exceeded? Otherwise it will just show up as an opaque deserialization error.

Reply:
It should be caught by this trace:
```
tracing::error!("RequestResponse outbound error for peer: {:?} with id: {:?} and error: {:?}", peer, request_id, error);
```